src/content/advanced/high-availability/multi-az/index.md

```diff
@@ -64,6 +66,43 @@ This enables use cases such as:
 
 Spreading worker nodes over multiple availability zones can be configured per [node pool]({{< relref "/advanced/node-pools" >}}) and independent of the choice of a single control plane node vs. using multiple control plane nodes (currently multiple control plane nodes are only supported on AWS).
 
+## Example Pod Topology Spread Constraints and Affinity
+
+To make sure your workload gets scheduled over available worker nodes over availability zones you can make use of [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) and [Affinity and anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity).
+
+In this example, the two constraints make sure, pods with the given labels are distributed across available zones. The first constraint makes sure the number of Pods across Nodes with the same `topology.kubernetes.io/zone` label is one at max. The second constraint picks the Node with the lowest number of Pods still across all nodes and all zones. This might collide with the first constraint and no schedule would happen at all unless `whenUnsatisfiable` is set to `ScheduleAnyway`. The `affinity` makes sure to not schedule any Pods on `master` Nodes.
+
+The following fields are part of the `spec.template.spec` of Deployments, StatefulSets, and DaemonSets:
+
+```
+topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/name: hello-world-app
+        app.kubernetes.io/version: 0.1.0
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/name: hello-world-app
+        app.kubernetes.io/version: 0.1.0
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/role
+          operator: NotIn
+          values:
+          - master
+```
+
+These constraints are checked during scheduling of new Pods. In case some dis-balancing happened because of other reasons, [Descheduler for Kubernetes](https://github.com/kubernetes-sigs/descheduler) can be used to re-schedule Pods to be more balanced.
+
```

Comment:
Maybe add something like this? Things can really be a bit tricky and it's good if people are aware.
```suggestion
As some undesirable edge-cases might show up on more complex setups please see the following issues for some discussions: [#106127](https://github.com/kubernetes/kubernetes/issues/106127) [#105977](https://github.com/kubernetes/kubernetes/issues/105977) [#105291](https://github.com/kubernetes/kubernetes/issues/105291).
```

Reply:
@webwurst updated your suggestion. Still okay?

Reply:
Yes, totally fine.